app/resources/v1/pacbio/pool_resource.rb

```diff
@@ -43,6 +43,12 @@ class PoolResource < JSONAPI::Resource
       # to republish the messages for the run
       after_update :publish_messages
 
+      # Filters
+      # join pool with tube as a pool has a barcode through tube
+      filter :barcode, apply: lambda { |records, value, _options|
```

Comment:
Might make sense to have this a scope (see plate resource). so it would be` records.by_barcode(barcode).`